src/sdk.ts

```diff
@@ -35,13 +37,15 @@ class KontentSmartLinkSDK {
     const level = this.configurationManager.debug ? LogLevel.Debug : LogLevel.Info;
     Logger.setLogLevel(level);
 
-    if (isInsideIFrame()) {
-      this.initializeIFrameCommunication();
-    } else if (this.configurationManager.queryParam) {
+    if (this.configurationManager.queryParam) {
       this.queryParamPresenceWatcher.watch(this.configurationManager.queryParam, this.nodeSmartLinkProvider.toggle);
     } else {
       this.nodeSmartLinkProvider.enable();
     }
+
+    if (isInsideIFrame()) {
```

Comment:
This `if` makes the `if (!isInsideIFrame()) {...}` inside `initializeIFrameCommunication()` unreachable.